packages/core/src/utils/result.ts

```diff
@@ -146,8 +150,12 @@ export const mergeResultPatch = (
       }
     }
   } else {
-    withData.data = nextResult.data || prevResult.data;
-    errors = (nextResult.errors as any[]) || errors;
+    withData.data = (nextResult.payload || nextResult).data || prevResult.data;
+    errors =
+      [
+        ...((nextResult.errors || []) as any[]),
+        ...((nextResult.payload && nextResult.payload.errors) || []),
+      ] || errors;
```

Comment:
I think the protocol assumes that `payload.errors` and `errors` are mutually exclusive 🤔 I think a protocol-level error will mean that `payload` isn't defined, looking at the spec again, so maybe we're fine with:

```suggestion
    errors =
      (nextResult.errors as any[]) ||
      (nextResult.payload && nextResult.payload.errors) || 
      errors;
```

This also solves the problem of us potentially creating an empty `errors` array here, even without any errors 🤔